plugins/chat/assets/javascripts/discourse/models/chat-channel.js

```diff
@@ -105,6 +116,57 @@ export default class ChatChannel extends RestModel {
     return this.currentUserMembership.following;
   }
 
+  get visibleMessages() {
+    return this.messages.filter((message) => message.visible);
+  }
+
+  set details(details) {
+    this.canDeleteOthers = details.can_delete_others ?? false;
+    this.canDeleteSelf = details.can_delete_self ?? false;
+    this.canFlag = details.can_flag ?? false;
+    this.canModerate = details.can_moderate ?? false;
+    if (details.can_load_more_future !== undefined) {
+      this.canLoadMoreFuture = details.can_load_more_future;
+    }
+    if (details.can_load_more_past !== undefined) {
+      this.canLoadMorePast = details.can_load_more_past;
+    }
+    this.userSilenced = details.user_silenced ?? false;
+    this.status = details.channel_status ?? "open";
```

Comment:
This should always have a value, no?

Reply:
Yes you must be right, will just remove it :+1: